### ISSUE TITLE
aspnet docker does not contain aspnet in 2.1

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/dotnet-docker#636]
_From @noelmcgrath on July 16, 2018 6:52_

In 2.0 Dockerfile a csproj was created and a dotnet restore done to bring down packages
See https://github.com/aspnet/aspnet-docker/blob/master/2.0/jessie/sdk/Dockerfile

this is no longer the case with 2.1 as it is "FROM microsoft/dotnet:2.1-sdk" and does not contain any aspnet packages

## Describe the solution you'd like
Just like aspdocker 2.0 Dockerfile include this step in sdk Dockerfile
     # warmup NuGet package cache

## Describe alternatives you've considered
Alternative is to create your own Dockerfile with the step in it

## Additional context
Create your own is a solution, but surely the aspnet docker sdk should come with aspnet in it

_Copied from original issue: aspnet/aspnet-docker#430_

**Comment by MichaelSimons** [comment]
@noelmcgrath, the SDK ships with a NuGetFallbackFolder that includes ASP.NET Core assemblies.  `/usr/share/dotnet/sdk/NuGetFallbackFolder` on Linux and `c:\Program Files\dotnet\sdk\NuGetFallbackFolder` on Windows. This is hooked up to NuGet so that if a package isn't found in the NuGet Cache it will check the fallback folder before downloading packages.  Can you explain your scenario in more detail?   What packages and versions are you depending on that aren't in the NuGetFallbackFolder?

**Comment by noelmcgrath** [comment]
@MichaelSimons  I have confirmed the fallback folder is there, however it does not seem to be used
When I run `dotnet restore  -v d` to get more output  I can see it is always calling out to api.nuget.org for packages

### Snippet at end of a csproj restore
We have a local nuget server for our own internal packages

NuGet Config files used:
	 /workspace/NuGet.Config
	 /root/.nuget/NuGet/NuGet.Config
 
 Feeds used:
	 https://nugetserver.ccs.local/nuget
	 https://api.nuget.org/v3/index.json
 
 Installed:
	 26 package(s) to /workspace/src/MYCOMPANYPROJECT.Domain.csproj
	 62 package(s) to /workspace/src/MYCOMPANYPROJECT.Infrastructure.csproj
	 62 package(s) to /workspace/src/MYCOMPANYPROJECT.Runner.csproj
	 86 package(s) to /workspace/test/MYCOMPANYPROJECT.Domain.UT.csproj
	 86 package(s) to /workspace/test/MYCOMPANYPROJECT.IT.csproj
	 86 package(s) to /workspace/test/CMYCOMPANYPROJECT.Infrastructure.UT.csproj
	 210 package(s) to /workspace/src/MYCOMPANYPROJECT.Web.csproj
	 234 package(s) to /workspace/test/MYCOMPANYPROJECT.Web.UT.csproj
Done executing task "RestoreTask".

###  Restoring packages for 1 of the csproj
Calls to local nuget server and api.nuget.org
	   
Restoring packages for /workspace/test/MYCOMPANYPROJECT.UT.csproj...
 Restoring packages for .NETCoreApp,Version=v2.1...
   GET https://nugetserver.ccs.local/nuget/FindPackagesById()?id='Microsoft.NET.Test.Sdk'&semVerLevel=2.0.0
   GET https://api.nuget.org/v3-flatcontainer/microsoft.net.test.sdk/index.json
   GET https://nugetserver.ccs.local/nuget/FindPackagesById()?id='NSubstitute'&semVerLevel=2.0.0
   GET https://api.nuget.org/v3-flatcontainer/nsubstitute/index.json
   GET https://nugetserver.ccs.local/nuget/FindPackagesById()?id='NUnit'&semVerLevel=2.0.0
   GET https://api.nuget.org/v3-flatcontainer/nunit/index.json
   GET https://nugetserver.ccs.local/nuget/FindPackagesById()?id='NUnit3TestAdapter'&semVerLevel=2.0.0
   GET https://api.nuget.org/v3-flatcontainer/nunit3testadapter/index.json
   OK https://nugetserver.ccs.local/nuget/FindPackagesById()?id='Microsoft.AspNet.WebApi.Client'&semVerLevel=2.0.0 250ms
   OK https://nugetserver.ccs.local/nuget/FindPackagesById()?id='Microsoft.AspNetCore'&semVerLevel=2.0.0 249ms
   OK https://nugetserver.ccs.local/nuget/FindPackagesById()?id='Microsoft.AspNetCore.Antiforgery'&semVerLevel=2.0.0 249ms
   OK https://nugetserver.ccs.local/nuget/FindPackagesById()?id='Microsoft.AspNetCore.Authentication.Abstractions'&semVerLevel=2.0.0 247ms
   OK https://nugetserver.ccs.local/nuget/FindPackagesById()?id='Microsoft.AspNetCore.Authentication'&semVerLevel=2.0.0 249ms
   OK https://nugetserver.ccs.local/nuget/FindPackagesById()?id='Microsoft.AspNetCore.Authentication.Cookies'&semVerLevel=2.0.0 248ms
   OK https://nugetserver.ccs.local/nuget/FindPackagesById()?id='Microsoft.AspNetCore.Authentication.Core'&semVerLevel=2.0.0 247ms
   OK https://api.nuget.org/v3-flatcontainer/microsoft.aspnet.webapi.client/index.json 256ms
   GET https://api.nuget.org/v3-flatcontainer/microsoft.aspnet.webapi.client/5.2.4/microsoft.aspnet.webapi.client.5.2.4.nupkg
   OK https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore/index.json 257ms
   GET https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore/2.1.0/microsoft.aspnetcore.2.1.0.nupkg
   OK https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication/index.json 255ms
   GET https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication/2.1.0/microsoft.aspnetcore.authentication.2.1.0.nupkg
   OK https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication.cookies/index.json 254ms
   GET https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication.cookies/2.1.0/microsoft.aspnetcore.authentication.cookies.2.1.0.nupkg
   OK https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication.microsoftaccount/index.json 252ms
   GET https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication.microsoftaccount/2.1.0/microsoft.aspnetcore.authentication.microsoftaccount.2.1.0.nupkg
   OK https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication.oauth/index.json 253ms
   GET https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication.oauth/2.1.0/microsoft.aspnetcore.authentication.oauth.2.1.0.nupkg
   OK https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication.google/index.json 255ms
   GET https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication.google/2.1.0/microsoft.aspnetcore.authentication.google.2.1.0.nupkg
   OK https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication.jwtbearer/index.json 255ms
   GET https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore.authentication.jwtbearer/2.1.0/microsoft.aspnetcore.authentication.jwtbearer.2.1.0.nupkg
   ETC .........

**Comment by noelmcgrath** [comment]
### My nuget config file
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="ccs" value="https://nugetserver.ccs.local/nuget" />
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
  </packageSources>
  <packageRestore>
    <add key="enabled" value="True" />
    <add key="automatic" value="True" />
  </packageRestore>
  <bindingRedirects>
    <add key="skip" value="False" />
  </bindingRedirects>
  <packageManagement>
    <add key="format" value="1" />
    <add key="disabled" value="True" />
  </packageManagement>
</configuration>
```

**Comment by bstack** [comment]
Hi I have the same issue, does not seem to use the fallback folder. Is there any solution to this?

Thanks,
Billy

**Comment by noelmcgrath** [comment]
Any update on this, calling out to nuget for packages is costing a lot of time when running ci\cd pipeline

**Comment by MichaelSimons** [comment]
@noelmcgrath -  In looking at the diagnostic log you included, it looks as though something in your project is referencing older versions of the packages that are not in the NuGetFallbackFolder included with the 2.1 SDK. 

```
OK https://nugetserver.ccs.local/nuget/FindPackagesById()?id='Microsoft.AspNet.WebApi.Client'&semVerLevel=2.0.0 250ms
...
GET https://api.nuget.org/v3-flatcontainer/microsoft.aspnetcore/2.1.0/microsoft.aspnetcore.2.1.0.nupkg  
```

It is by design that the NuGetFallbackFolder only contains the current packages.

**Comment by MichaelSimons** [comment]
Closing - NuGetFallbackFolder does contain the ASP.NET Core packages.  As noted in this particular case, the NuGet packages being referenced are referring to older versions which is why packages are getting pulled down.
